gp2gp-translator/src/test/java/uk/nhs/adaptors/pss/translator/mapper/DuplicateObservationStatementMapperTest.java

```diff
@@ -229,37 +259,61 @@ public void doesntMergeObservationWhereTheMatchingObservationHasMultiplePertinen
             generateLinksetComponent("ID-1")
         ));
 
-        mapper.mergeDuplicateObservationStatements(ehrExtract);
+        mapper.mergeOrRemoveDuplicateObservationStatements(ehrExtract);
 
         assertThat(firstEhrComposition(ehrExtract)).hasSize(3);
     }
 
     @Test
-    public void doesntMergeObservationWhereTheObservationDoesNotEndInEllipses() {
+    public void doesntMergeObservationWhereTheObservationDoesNotEndInEllipsesAndCodeableConceptCodesAreDifferent() {
         var ehrExtract = createExtract(List.of(
                 createObservation("ID-1", "101", "This is an observation doesnt end with ellipses:::"),
-                createObservation("ID-2", "101", "This is an observation which ends with ellipses removed."),
+                createObservation("ID-2", "102", "This is an observation which ends with ellipses removed."),
                 generateLinksetComponent("ID-1")
         ));
 
-        mapper.mergeDuplicateObservationStatements(ehrExtract);
+        mapper.mergeOrRemoveDuplicateObservationStatements(ehrExtract);
 
         assertThat(firstEhrComposition(ehrExtract)).hasSize(3);
     }
 
     @Test
-    public void doesntMergeObservationWhereTheObservationIsLessThan47Chars() {
+    public void removeObservationWhereTheObservationDoesNotEndInEllipsesButCodeableConceptCodesAreTheSame() {
```

Comment:
Hmm, is this a requirement?

Reply:
I've had a look at the XML provided within the ticket, and my interpretation of the requirements are slightly different. I don't think an Observation should be removed where there are diverging bits of text, or if they are, then there needs to be some merging behaviour to ensure that any text doesn't get removed.